flang/lib/Lower/Allocatable.cpp

```diff
@@ -454,7 +454,9 @@ class AllocateStmtHelper {
                            const fir::MutableBoxValue &box) {
     if (!box.isDerived() && !errorManager.hasStatSpec() &&
         !alloc.type.IsPolymorphic() && !alloc.hasCoarraySpec() &&
-        !useAllocateRuntime) {
+        !useAllocateRuntime && !box.isPointer()) {
```

Comment:
Should there be the same check in `genDeallocate` below?

Reply:
The bad deallocations are being caught in my testing without it, but I don't really understand the flow here.  Should I just add the check to `genDeallocate()` anyway?

Reply:
I bet it's the presence of `stat=` in my tests that makes the difference.  `PointerDeallocate()` should be called even without `stat=`.  Will update.

Reply:
https://github.com/llvm/llvm-project/pull/79702